Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Transformer.purs

```diff
@@ -68,12 +68,12 @@ buildTransactionDetails (API.HistoryEntryDetailsEntityV2Resp resp) gradientConfi
                   {
                     key : "FEE_BREAKUP",
                     title : getString FEE_BREAKUP,
-                    val : if (resp.feeType == AUTOPAY_REGISTRATION || not showFeeBreakup) then "" else (getFeeBreakup offerAndPlanDetails driverFee'.totalRides <> " "<> getString GST_INCLUDE)
+                    val : if (resp.feeType == AUTOPAY_REGISTRATION || not showFeeBreakup) then "" else (getFeeBreakup driverFee'.maxRidesEligibleForCharge (driverFee'.planAmount - (fromMaybe 0.0 driverFee'.totalSpecialZoneCharges)) driverFee'.totalRides)
                   },
                   {
                     key : "BOOTH_CHARGES",
                     title : getString BOOTH_CHARGES,
-                    val : case driverFee'.specialZoneRideCount, driverFee'.specialZoneAmount of
+                    val : case driverFee'.specialZoneRideCount, driverFee'.totalSpecialZoneCharges of
```

Comment:
case driverFee'.specialZoneRideCount, driverFee'.totalSpecialZoneCharges of Just 0, Just 0.0 -> "" Just count, Just charges -> show count <> " " <>getString RIDES <> " x ₹" <> getFixedTwoDecimals (charges / INT.toNumber count) <> " " <> getString GST_INCLUDE

if (resp.feeType == AUTOPAY_REGISTRATION || not showFeeBreakup) then "" else (getFeeBreakup driverFee'.maxRidesEligibleForCharge (driverFee'.planAmount - (fromMaybe 0.0 driverFee'.totalSpecialZoneCharges)) driverFee'.totalRides)

move these part to the top let declarations as it is used in two places.